lib/kafkaClient.js

```diff
@@ -604,7 +666,10 @@ KafkaClient.prototype.createBroker = function (host, port, longpolling) {
     } else {
       self.initializeBroker(brokerWrapper, function (error) {
         if (error) {
-          logger.error('error initialize broker after connect', error);
+          logger.error('error initializing broker after connect', error);
+          if (error instanceof errors.SaslAuthenticationError) {
+            self.emit('auth_error', error);
```

Comment:
Great question, I think I was concerned about raising an `error` here when we don't do so for other types of error (and presumably raising other types of error here would have knock-on implications for tests, client code, etc.)

To put it another way, I wanted to bubble up this error for test purposes but wasn't convinced this PR warranted changing `error` semantics & potentially breaking stuff. If that makes any sense at all. It makes a potentially large/risky change even larger and riskier. 😅

Reply:
I'm leaning towards emitting an error. I would expect consumers/producers would want to be notified of this failure since a failure in authentication means messages will not be received or delivered.